src/java.base/share/classes/sun/util/calendar/AbstractCalendar.java

```diff
@@ -165,31 +165,18 @@ public long getTime(CalendarDate date) {
             if (date.isNormalized()) {
                 return ms - date.getZoneOffset();
             }
+
             // adjust time zone and daylight saving
-            int[] offsets = new int[2];
-            if (date.isStandardTime()) {
-                // 1) 2:30am during starting-DST transition is
-                //    intrepreted as 2:30am ST
-                // 2) 5:00pm during DST is still interpreted as 5:00pm ST
-                // 3) 1:30am during ending-DST transition is interpreted
-                //    as 1:30am ST (after transition)
-                if (zi instanceof ZoneInfo) {
-                    ((ZoneInfo)zi).getOffsetsByStandard(ms, offsets);
-                    zoneOffset = offsets[0];
-                } else {
-                    zoneOffset = zi.getOffset(ms - zi.getRawOffset());
-                }
+            // 1) 2:30am during starting-DST transition is
+            //    intrepreted as 3:30am DT
+            // 2) 5:00pm during DST is intrepreted as 5:00pm DT
+            // 3) 1:30am during ending-DST transition is interpreted
+            //    as 1:30am DT/0:30am ST (before transition)
+            if (zi instanceof ZoneInfo zInfo) {
+                // Offset value adjusts accordingly depending on DST status of date
```

Comment:
Historically, this `if else` has not been touched since the introduction of the class.

The original code has a structure that one can presume follows the logic, if `isStandardTime()`, get a standard offset, otherwise get a day light saving offset. This is not the case. 

The code within the `else` statement is able to retrieve the correct offset if the date is in standard **or** in day light saving time (not just a day light saving offset, as the original code would imply). Consider the following example,

```
// Where ms is calculated from the date: LA time zone at 3-13-2016 at 4 AM (daylight saving)
zoneOffset = zInfo.getOffsetsByWall(ms, new int[2]); 
// returns the adjusted offset, -25200000 (7 hours)

// Where ms is calculated from the date: LA time zone at 1-13-2016 at 4 AM (standard)
zoneOffset = zInfo.getOffsetsByWall(ms, new int[2]); 
// returns the standard offset, -28800000 (8 hours)
```
Removing this code is not only safe because `isStandardTime()` is always `false`
- tiers 1-3 clean
- breakpoint within the original `if` never breaks execution for JDK Calendar tests

But we can also feel that the change is not suspicious since the code within the `else` block can produce a standard or daylight offset.

Reply:
I suspect that the original design is to have this overridden by possible subclasses if needed, but none has done so far (and I don't think it ever will). So I think it is safe (and less complex) to remove these unused codes.